pkg/asset/machines/master.go

```diff
@@ -606,3 +609,31 @@ func IsMachineManifest(file *asset.File) bool {
 		return matched
 	}
 }
+
+func generateAsset(objects interface{}, fileName string) (assets []*asset.File, err error) {
+	switch reflect.TypeOf(objects).Kind() {
```

Comment:
Do not use reflection as a means to avoid code duplication. Either create a `[]interface{}` to pass into the function or have the function operator on each element individually. I realize that there will still be quite a bit of boilerplate code with either approach, but that is go.

Reply:
Unfortunately it's not possible to use `[]interface{}` as function parameter as [disallowed](https://go.dev/doc/faq#convert_slice_of_interface) by the language. I've thus kept the conversion code within the `generateAsset` boundaries to minimize the noise outside, but that small portion cannot be simplified further, so I guess this could be a reasonable compromise between not using reflection and removing code duplication (while operating on each element individually appeared to make the code more complicated). I guess that in future Type Parameters could help for that.

Reply:
That is exactly what I was trying to suggest with "create a `[]interface{}` to pass into the function".